### ISSUE TITLE
Extension clobbers global gitconfig setting

### ISSUE THREAD

**Issue by AjayKMehta** [hediet/vscode-hediet-power-tools#10]
I was wondering why the `core.excludesFile` setting in my global .gitconfig file was getting clobbered and I tracked it down to [this](https://github.com/hediet/vscode-hediet-power-tools/blob/b5b68009d6dbf4d1bbdb49f27330ec3e8eea6e1f/src/features/MarkdownEditorProjection/MarkdownDocumentController.ts#L70-L80).

I had the value in the file set to `C:\\users\\ajay\\.gitconfig`. 

`git config --get core.excludesFile` returns `c:\users\ajay\.gitconfig`.

The issue I believe is that `existsSync` does not handle Window-style paths correctly (not verified). I also saw that [it does not handle `~` in file path](https://stackoverflow.com/questions/19384256/node-js-fs-existssync-returns-incorrect-value-for-paths-containing-environment-v).

Can you please fix this so it handles these cases correctly? For now, I switched to Unix style path to be safe.

Thanks.

**Comment by hediet** [comment]
Thanks for reporting this!

What is the expected value of `core.excludesFile` and what is the actual value?

What is your suggestion to fix this?

**Comment by AjayKMehta** [comment]
So, the value in the .gitconfig file has `\\` as path separator but as stated in my previous comment, `git config --get core.excludesFile` returns **c:\users\ajay\.gitconfig** with `\` as separator which `existsSync` does not like.

Screenshot below illustrates this as well as issue with `~`:

![existsSync](https://user-images.githubusercontent.com/11180071/153472719-f601150d-020c-4486-9f68-c3fc75bd5d6a.png)

My suggestion is to replace single `\` with `\\` before calling this method, replace `~` using advice from [accepted answer here](https://stackoverflow.com/questions/19384256/node-js-fs-existssync-returns-incorrect-value-for-paths-containing-environment-v) or use a different library.

I'm not really familiar with JavaScript or TypeScript or else, I would have happily submitted a PR to fix this 😀 

Thank you for making such a great extension that I use daily!

**Comment by hediet** [comment]
> with \ as separator which existsSync does not like.

This cannot be the issue.

When you provide a string literal in JS, you just have to double enquote `\`.
`\u` starts a unicode sequence, but digits have to follow.

I.e. `"C:\\users"` literally represents the string `C:\users`.

**Comment by AjayKMehta** [comment]
Yes, you are right. I should have been **more specific** saying that single `\` is a problem when followed by certain characters such that it forms an escape sequence.

For example, `\n` is also a problem:

`fs.existsSync("C:\program files\notepad++")` returns `false`.
`fs.existsSync("C:\\program files\\notepad++")` returns `true`.

Solution is to replace single `\` with `\\` or `/` in return value of `git config --get core.ExcludesFile`.

**Comment by Bobronium** [comment]
I have the same issue. Even when I'm editing `.git-global-excludes-file` it's still gets replaced every time. I'm on macOS.

@hediet, I think the issue comes down to 
1. not stripping stdout, so `fs.existsSync("~/.gitignore\n")` is called
2. as @AjayKMehta mentioned, `fs.existsSync` doesn't expand `~/` in path

So solution would be:
1. call `stdout.trim()`
2. expand `~/` if found in path

<details>

```js
$ node
Welcome to Node.js v18.2.0.
Type ".help" for more information.
> const { promisify } = require('util');
undefined
> const { exec } = require('child_process');
undefined
> const { stdout } = await promisify(exec)(
... 			"git config --get core.excludesfile"
... 		)
undefined
> stdout
'~/.gitignore\n'
>
```
</details>
@AjayKMehta, you seem to misunderstand how escaping works.

I'm familiar with python, so will write an example using it, however same applies to JS, AFAIK.

When you hardcode strings, you have to escape certain characters because they would be replaced by interpreter otherwise:
```py
var = "C:\\program files\\notepad++"
print(var)  # C:\program files\notepad++
```
But when you're dealing with input, you don't have to worry about, say, `\n` sequences, because it would literally represent a `\` followed by `n`, same as you would get by hardcoding `"\\n"`:
```py
var = input()
print(var)  # C:\program files\notepad++
```

**Comment by hediet** [comment]
@Bobronium thanks for your analysis, can you do a PR? :)

**Comment by AjayKMehta** [comment]
Thanks for the primer on escaping, @Bobronium.

I understand how escaping works but not how it is done in JS/TS 😃

A PR to fix this would be great.

**Comment by Bobronium** [comment]
> A PR to fix this would be great.

There is one!

**Comment by AjayKMehta** [comment]
Awesome 👍
